### PR TITLE
Declarative + activeData

### PR DESCRIPTION
**Pull request by stephendonner**
@davehunt / @rbillings /cc @mozilla/firefox-test-engineering r?

Mostly-passing (same as prod) ad-hoc run here: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/treeherder.adhoc/114/console

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,154 +1,64 @@
...
-            runTox(environment)
-          } catch(err) {
-            currentBuild.result = 'FAILURE'
-            throw err
-          } finally {
-            dir('results') {
-              stash environment
-            }
-          }
+    stage('Test') {
```
Comment (stephendonner) [inline Jenkinsfile]: Guessing this is some funky line-ending or editor-spacing/tabs vs. spaces thing; will try to fix.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,154 +1,64 @@
...
+      environment {
+        /** See https://issues.jenkins-ci.org/browse/JENKINS-42857 - we'd like to expand this out into multi-line concatenations */
+        PYTEST_ADDOPTS = "--color=yes --driver=SauceLabs --variables=capabilities.json"
```
Comment (davehunt) [inline Jenkinsfile]: Put this in the main `environment` section on line 19. We only need to have two sections when we're using a secret variables file.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,154 +1,64 @@
...
-            currentBuild.result = 'FAILURE'
-            throw err
-          } finally {
-            dir('results') {
-              stash environment
-            }
-          }
+    stage('Test') {
+      environment {
+        /** See https://issues.jenkins-ci.org/browse/JENKINS-42857 - we'd like to expand this out into multi-line concatenations */
```
Comment (davehunt) [inline Jenkinsfile]: As mentioned with previous pull requests, let's fix the issue number here if we want to restore the multiline string in the future, though I think in a few days we may be able to revert this workaround.

File: Jenkinsfile (hunk 2 above)
Comment (davehunt) [inline Jenkinsfile]: Also please add `--tb=short`

**Pull request merged**